### PR TITLE
[NHR-112] DonationPage static styles

### PR DESCRIPTION
**Pull request by michael-caktus**
This includes implementing a bunch of frontend elements that respond to the `elements` field on DonationPage. At the bottom here is an example of some valid JSON you can plop in there to test. 

The meat of this PR starts at `spa/src/components/donationPage/DonationPage.js`. Before that, you'll see the following:
1. Change `DonationPage.title` to `DonationPage.heading` and add `DonationPage.graphic` for an image.
2. Fix a django urlpatterns for static assets error in local development 
3. Add a google font
4. Update frontend testing fixture

Starting around `spa/src/components/donationPage/DonationPage.js`, you'll see the more relevant changes. The donation page elements are broken down in to two categories. Static and Dynamic. Components starting with S are static and D are dynamic. Their imported through sort of dynamic/static index files and fetched by name. Eventually that name, as well as the UUID field in the JSON below, will also be vital for the editing flow.

This work also includes a very basic implementation of Draft.js rich-text-editor. This was necessary so we could see what kind of output we can get from it.  Turns out we can, with the help of a plugin, get HTML. Which is great.

The DonationPage.elements JSON:
```
[
    {"type": "DRichText", "uuid": "testRichText1", "content": "<p><strong>Your support keeps us going! Become a member today.</strong></p> <p>Because of your generosity, we have the courage, freedom, and independence to dedicate our entire newsroom to telling stories that spark action, improve lives and protect our democracy. Thank you for your support that keeps us going.</p>"}, 
    {"type": "DFrequency", "uuid": "testFreq1", "content": [{"value": "one_time", "displayName": "One time"}, {"value": "month", "displayName": "Monthly"}, {"value": "year", "displayName": "Yearly"}]},
    {"type": "DAmount", "uuid": "testAmount1", "content": {"offerPayFees": true, "allowOther": true, "options": {
        "one_time": [120, 180, 365],
        "month": [10, 15, 25],
        "year": [120, 180, 365]
      }}
    },
    {"type": "DDonorInfo", "uuid": "testDonorInfo1"},
    {"type": "DAdditionalInfo", "uuid": "testAdditinoalInfo", "content": [
      {"type": "text", "label": "I'm contributing because...", "name": "reason"},
      {"type": "text", "label": "Why does anybody even do anythign?", "name": "purpose"}
    ]}
  ]
```

### CODE REVIEW

File: spa/src/elements/inputs/RichTextEditor.js [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+//   if (e.keyCode === 9 /* TAB */) {
+//     const newEditorState = RichUtils.onTab(e, this.state.editorState, 4 /* maxDepth */);
+//     if (newEditorState !== this.state.editorState) {
+//       this.onChange(newEditorState);
+//     }
+//     return;
+//   }
+//   return getDefaultKeyBinding(e);
+// }
+
```
Comment (benjaminEwhite) [inline spa/src/elements/inputs/RichTextEditor.js:115]: Remove

File: spa/src/elements/inputs/RichTextEditor.js [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+//           <Editor
+//             blockStyleFn={getBlockStyle}
+//             customStyleMap={styleMap}
+//             editorState={editorState}
+//             handleKeyCommand={this.handleKeyCommand}
+//             keyBindingFn={this.mapKeyToEditorCommand}
+//             onChange={this.onChange}
+//             placeholder="Tell a story..."
+//             ref="editor"
+//             spellCheck={true}
```
Comment (benjaminEwhite) [inline spa/src/elements/inputs/RichTextEditor.js:151]: Remove

File: spa/src/elements/inputs/RichTextEditor.js [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+//           />
+//         </div>
+//       </div>
```
Comment (benjaminEwhite) [inline spa/src/elements/inputs/RichTextEditor.js:154]: Remove (couldn't easily select all the lines in one pass, fwiw).

File: spa/src/elements/inputs/RichTextEditor.js [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+
+//   render() {
+//     return (
+//       <span className={className} onMouseDown={this.onToggle}>
+//         {this.props.label}
+//       </span>
+//     );
+//   }
+// }
+
```
Comment (benjaminEwhite) [inline spa/src/elements/inputs/RichTextEditor.js:218]: Remove

File: spa/src/elements/inputs/RichTextEditor.js [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+          label={type.label}
+          onToggle={onToggle}
+          style={type.style}
+        />
+      ))}
+    </div>
+  );
+};
+
+var INLINE_STYLES = [
```
Comment (benjaminEwhite) [inline spa/src/elements/inputs/RichTextEditor.js:248]: Why not `const` (or barring that, `let`)?

File: spa/src/components/donationPage/pageGetters.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  return <staticElements.SPageHeading />;
+};
+
+export const getGraphicElement = () => {
+  return <staticElements.SGraphic />;
+};
+
+export const getBenefitsElement = () => {
+  return <staticElements.SBenefits />;
+};
```
Comment (benjaminEwhite) [inline spa/src/components/donationPage/pageGetters.js:18]: Just curious for reasoning behind having these as non-dynamic functions that will always return a given component? is it so there's a homogeneous API?

**Review by benjaminEwhite — COMMENTED** [review]
Some comments and small amount of cleanup to do, but otherwise this is looking good. FWIW, I only reviewed code and didn't test out in local.

File: spa/src/components/donationPage/pageContent/DAmount.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            />
+            <span>{getFrequencyRate(frequency)}</span>
+          </S.OtherAmount>
+        )}
+      </S.DAmount>
+      {element.content.offerPayFees && (
+        <S.PayFees>
+          <S.PayFeesQQ>Agree to pay fees?</S.PayFeesQQ>
+          <S.Checkbox
+            label={`${'$2.99'} ${getFrequencyAdverb(frequency)}`}
```
[the remark this answers is not in the document]
Comment (michael-caktus) [reply spa/src/components/donationPage/pageContent/DAmount.js:66]: This is just a stand-in until the dynamic fee is implemented

File: spa/src/components/donationPage/pageContent/dynamicElements.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export { default as DRichText } from './DRichText';
```
[the remark this answers is not in the document]
Comment (michael-caktus) [reply spa/src/components/donationPage/pageContent/dynamicElements.js:1]: We don't always know the shape of props for these elements, so we can't generalize at this point.

File: spa/src/elements/inputs/RichTextEditor.js (hunk 5 above)
Comment (michael-caktus) [reply spa/src/elements/inputs/RichTextEditor.js:248]: oops, copy pasta!

File: spa/src/components/donationPage/pageGetters.js (hunk 6 above)
Comment (michael-caktus) [reply spa/src/components/donationPage/pageGetters.js:18]: Yeah, that was the idea. When you put it that way, though, it does seem a little strange. I'll look at the context a bit and see if I want to change those.

**Comment by michael-caktus** [comment]
Latest commit removes some commented code and adds some documentation. I've left the RichText stuff alone, because a later PR wipes those files out.

**Comment by michael-caktus** [comment]
Cypress test failure is due to the github action config, not a code failure.

The issue is fixed in a subsequent PR.

Moving this one ahead.

**Pull request merged**